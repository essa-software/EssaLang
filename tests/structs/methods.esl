struct Foo {
    field: u32;

    func bar(this, a: u32) {
        print("Hello from foo.bar()! a={} field={}", a, this.field);
    }
}

func main() {
    mut foo: Foo = Foo { field: 2137 };
    foo.bar(52);
}
