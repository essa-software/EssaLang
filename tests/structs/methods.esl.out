Hello from foo.bar()! a=52 field=2137